platform-coop-toolkit/coop-library-framework
# Rework array and object metadata registration

## Description

This PR adds support for WordPress 5.3's new object and array meta types.

## Steps to test

1. Ensure that your development environment is running [WordPress 5.3-beta](https://wordpress.org/news/2019/09/wordpress-5-3-beta-2/):
```
wp core update --version=5.3-beta2
```
2. Create a resource with archival links, authors, and revisions.
3. View the resource in the REST API: `/wp-json/wp/v2/resources/[ID]`

**Expected behavior:** Response JSON should include:

```js
{
  "meta": {
    "lc_resource_wayback_machine_links": [
      "https://web.archive.org/https://example.com"
    ],
    "lc_resource_author": [
      "Author One",
      "Author Two"
    ],
    "lc_resource_revisions": [
      {
        "revision_date": "2019-10-07",
        "revision_description": "Revision description."
      }
    ]
  }
}
```

## Additional information

- [WP 5.3 Supports Object and Array Meta Types in the REST API](https://make.wordpress.org/core/2019/10/03/wp-5-3-supports-object-and-array-meta-types-in-the-rest-api/)

## Related issues

Not applicable.
